graspologic/embed/case.py

```diff
@@ -0,0 +1,314 @@
+            Desired dimensionality of output data. If "full",
+            n_components must be <= min(X.shape). Otherwise, n_components must be
+            < min(X.shape). If None, then optimal dimensions will be chosen by
+            ``select_dimension`` using ``n_elbows`` argument.
+
+        alpha : float, optional (default = None)
+            Tuning parameter to use. Not used if embedding_alg == cca:
+                -  None: tune the alpha-value by minimizing the k-means objective function.
+                         Since this involves running k-means over a parameter space, this
+                         will result in a slower algorithm in exchange for likely better
+                         clustering.
+                - float: use a particular alpha-value. Results in a much faster algorithm
+                         (since we are not tuning with kmeans) in exchange for potentially
+                         suboptimal clustering results.
+                -    -1: Default to the ratio of the leading eigenvector of the Laplacian
+                         to the leading eigenvector of the covariate matrix. This will
+                         result in suboptimal clustering in exchange for increased
+                         clustering speed.
+
+        tuning_runs : int, optional (default = 100)
+            If tuning alpha with k-means, this parameter determines the number of times
+            k-means is run. Higher values are more computationally expensive in exchange
+            for a finer-grained search of the parameter space (and better embedding).
+
+        n_jobs : int, optional (default = None)
+            The number of parallel threads to use in K-means when calculating alpha.
+            `None` or `-1` means using all processors.
+
+        verbose : int, optional (default = 0)
+            Verbosity mode.
+
+        n_elbows : int, optional, default: 2
+            If `n_components=None`, then compute the optimal embedding dimension using
+            `select_dimension`. Otherwise, ignored.
+
+        check_lcc : bool , optional (defult =True)
+            Whether to check if input graph is connected. May result in non-optimal
+            results if the graph is unconnected. Not checking for connectedness may
+            result in faster computation.
+
+        concat : bool, optional (default = False)
+            If graph(s) are directed, whether to concatenate each graph's left and right
+            (out and in) latent positions along axis 1.
+
+
+        References
+        ---------
+        .. [1] Binkiewicz, N., Vogelstein, J. T., & Rohe, K. (2017). Covariate-assisted
+        spectral clustering. Biometrika, 104(2), 361-377.
+    """
+
+    def __init__(
+        self,
+        n_components=None,
+        embedding_alg="assortative",
+        alpha=None,
+        tuning_runs=20,  # TODO: this default is gonna take hella long on bigger matrices. also the one in R was 100
```

Comment:
maybe have a sliding default depending on the users size?

Reply:
irrelevant now cause I ripped out the tuning algorithm because it was annoying and usually didn't actually improve anything over just using the leading eigenvalues (and in fact often made the embedding worse)